scripts/kernel-features.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import tarfile
+import json
+import sys
+
+
+g_btf_info_config = b'CONFIG_DEBUG_INFO_BTF=y'
+g_all_search_options = [
+    g_btf_info_config
+]
+
+
+class Bundle:
+    def __init__(self, path):
+        self.path = path
+        self.filename = os.path.basename(path)
+        bundle, _ = os.path.splitext(self.filename)
+        self.version = bundle[len('bundle-'):]
+
+        self.btf = False
+
+    def find_features(self):
+        """
+        Searches the bundle for all known features, first searching the config,
+        and then in the wider bundle.
+        """
+        search_options = []
+        with tarfile.open(self.path) as tar:
+            try:
+                config = tar.extractfile('./.config')
+                config = config.read()
+
+                self.btf = g_btf_info_config in config
```

Comment:
The `KeyError` comes from line 33, where we are attempting to extract the config file from the archive. If that exists, then we are simply checking its contents (and each check of `thing in config` is safe). If it doesn't exist then we search the rest of the archive.

there is perhaps still an argument for tighter parsing of the value, but it might be a fair bit slower and at this level might be overkill because the config format is relatively strict and well-formed.